Projects/Server/Items/Container.Enumerable.cs

```diff
@@ -0,0 +1,226 @@
+
+    public AllItemsEnumerable<TFilter, TResult> GetAllItems<TFilter, TResult>(in TFilter filter = default)
+        where TFilter : struct, IFilter<Item, TResult>
+        => new(this, filter);
+
+    public AllItemsEnumerable<PredicateFilter<Item>, Item> GetAllItems(Predicate<Item> predicate)
+        => new(this, new(predicate));
+
+    public AllItemsEnumerable<TypeFilter<Item, TResult>, TResult> GetAllItems<TResult>()
+        => new(this, default);
+
+    public AllItemsEnumerable<TypeFilter<Item>, Item> GetAllItems(Type type)
+        => new(this, new(type));
+
+    public readonly ref struct AllItemsEnumerable<TFilter, TResult>
+        where TFilter : struct, IFilter<Item, TResult>
+    {
+        private readonly Container container;
+        private readonly ref TFilter filter;
+
+        public AllItemsEnumerable(Container container, scoped in TFilter filter = default)
+        {
+            this.container = container;
+            this.filter = filter;
+        }
+
+        public AllItemsEnumerable<ChainFilter<Item, TResult, TTResult, TFilter, TTFilter>, TTResult> Where<TTFilter, TTResult>(TTFilter filter = default)
+            where TTFilter : struct, IFilter<TResult, TTResult>
+        {
+            return new(container, new(this.filter, filter));
+        }
+
+        public AllItemsEnumerable<ChainFilter<Item, TResult, TTResult, TFilter, TypeFilter<TResult, TTResult>>, TTResult> OfType<TTResult>()
+        {
+            return new(container, new(filter, default));
+        }
+
+        public AllItemsEnumerable<ChainFilter<Item, TResult, TResult, TFilter, TakeFilter<TResult>>, TResult> Take(int amount)
+        {
+            return new(container, new(filter, new(amount)));
+        }
+
+        public TResult? FirstOrDefault(TResult? @default = default)
+        {
+            if (container.m_Items is not { Count: > 0 })
+            {
+                return @default;
+            }
+
+            Container? current = container;
+
+            do
+            {
+                Span<Item> itemsSpan = CollectionsMarshal.AsSpan(current.m_Items);
+
+                for (int i = 0; i < itemsSpan.Length; i++)
+                {
+                    Item item = itemsSpan[i];
+
+                    if (item is TResult tItem)
+                    {
+                        containersForGetItem.Clear();
+                        return tItem;
+                    }
+
+                    if (item is Container { m_Items.Count: > 0 } c)
+                    {
+                        containersForGetItem.Enqueue(c);
+                    }
+                }
+            }
+            while (containersForGetItem.TryDequeue(out current));
+
+            containersForGetItem.Clear();
+
+            return @default;
+        }
+
+        public bool Any() => FirstOrDefault() != null;
+
+        public PooledRefQueue<TResult> ToPooledRefQueue()
+        {
+            PooledRefQueue<TResult> toRet = new();
```

Comment:
We don't support `new()` for this ref queue. Change to `PooledRefQueue.Create()`

Reply:
My bad, I rewrote those code too many times that I forgot to use the Create method 😄

Reply:
Should be fixed now